networkit/cpp/generators/MocnikGenerator.cpp

```diff
@@ -0,0 +1,309 @@
+/*
+ * MocnikGenerator.cpp
+ *
+ *  Created on: July 7, 2018
+ *      Author: Franz-Benjamin Mocnik <mail@mocnik-science.net>
+ */
+
+#include "MocnikGenerator.h"
+#include "../auxiliary/Random.h"
+#include <cmath>
+#include <unordered_map>
+
+namespace NetworKit {
+
+MocnikGenerator::MocnikGenerator(count dim, count n, double k, bool weighted): dim(dim), weighted(weighted) {
```

Comment:
Please use tabs instead of spaces. Historically, there were different styles of formatting in NetworKit but we're trying to converge to tabs only.

Reply:
Corrected in another commit. It was neither clear from the existing code and nor from the [documentation](https://github.com/kit-parco/networkit/blob/Dev/Doc/DevGuide.mdown) whether to use spaces or tabs.

Reply:
Yes, sorry, the documentation is not clear on that; we will add that to the documentation.

Reply:
No problem. In general, I am happy to have the documentation. ;)